WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java

```diff
@@ -878,36 +880,38 @@ public void onClick(View v) {
 
             if (mPost.hasBlogName()) {
                 txtBlogName.setText(mPost.getBlogName());
-                txtBlogName.setVisibility(View.VISIBLE);
-            } else if (mPost.hasBlogUrl()) {
-                txtBlogName.setText(UrlUtils.getHost(mPost.getBlogUrl()));
-                txtBlogName.setVisibility(View.VISIBLE);
+            } else if (mPost.hasAuthorName()) {
+                txtBlogName.setText(mPost.getAuthorName());
             } else {
-                txtBlogName.setVisibility(View.GONE);
+                txtBlogName.setText(null);
             }
 
-            int avatarSz = getResources().getDimensionPixelSize(R.dimen.avatar_sz_medium);
             if (mPost.hasBlogUrl()) {
-                String imageUrl = GravatarUtils.blavatarFromUrl(mPost.getBlogUrl(), avatarSz);
-                imgAvatar.setImageUrl(imageUrl, WPNetworkImageView.ImageType.BLAVATAR);
+                int blavatarSz = getResources().getDimensionPixelSize(R.dimen.avatar_sz_medium);
+                String imageUrl = GravatarUtils.blavatarFromUrl(mPost.getBlogUrl(), blavatarSz);
+                imgBlavatar.setImageUrl(imageUrl, WPNetworkImageView.ImageType.BLAVATAR);
+                txtDomain.setText(UrlUtils.getHost(mPost.getBlogUrl()));
             } else {
-                imgAvatar.setImageUrl(mPost.getPostAvatarForDisplay(avatarSz), WPNetworkImageView.ImageType.AVATAR);
+                imgBlavatar.showDefaultBlavatarImage();
+                txtDomain.setText(null);
             }
 
-            if (mPost.hasAuthorName()) {
-                txtAuthor.setText(mPost.getAuthorName());
-                txtAuthor.setVisibility(View.VISIBLE);
+            if (mPost.hasPostAvatar()) {
+                int avatarSz = getResources().getDimensionPixelSize(R.dimen.avatar_sz_tiny);
+                imgAvatar.setVisibility(View.VISIBLE);
+                imgAvatar.setImageUrl(mPost.getPostAvatarForDisplay(avatarSz), WPNetworkImageView.ImageType.AVATAR);
             } else {
-                txtAuthor.setVisibility(View.GONE);
+                imgAvatar.setVisibility(View.GONE);
```

Comment:
When the avatar is missing, should we just show a default avatar resource? I think we handle this case on the server-side and return a default avatar url and it _never_ reaches to the _else_ statement, but I am not sure. If it ever does, it might be better to have that default avatar as a resource file. What do you think?